seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumComJSRunner.scala

```diff
@@ -51,7 +61,18 @@ class SeleniumComJSRunner(browserProvider: SeleniumBrowser,
 
         case msg: String =>
           browser.processConsoleLogs(console)
-          msg
+          if (msg.startsWith(MESSAGE_TAG)) {
+            val taglessMsg = msg.substring(MESSAGE_TAG.length)
+            taglessMsg.replace("\u0007\u0007", "\u0000").map {
+              case '\u0000' => '\u0007'
+              case '\u0007' => '\u0000'
+              case c        => c
+            }
```

Comment:
Arf, yes. That said, I don't quite understand how the current code is supposed to work. Maybe a good old loop/tailrec would be easier to get right.